### PR TITLE
[video_player] Calls `onDestroy` instead of `initialize` in onDetachedFromEngine

### PR DESCRIPTION
**Pull request by 0xZOne**
In the `onDetachedFromEngine` function, the `onDestroy` method should be called instead of the `initialize` method, even though the implementation of the two functions is no different.

```java
  private void onDestroy() {
    // The whole FlutterView is being destroyed. Here we release resources acquired for all
    // instances
    // of VideoPlayer. Once https://github.com/flutter/flutter/issues/19358 is resolved this may
    // be replaced with just asserting that videoPlayers.isEmpty().
    // https://github.com/flutter/flutter/issues/20989 tracks this.
    disposeAllPlayers();
  }

  public void initialize() {
    disposeAllPlayers();
  }
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change

File: packages/video_player/video_player_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ## NEXT
 
+## 2.4.13
 * Updates minimum supported SDK version to Flutter 3.13/Dart 3.1.
+* Call `onDestroy` instead of `initialize` in onDetachedFromEngine.
```
Comment (stuartmorgan) [inline packages/video_player/video_player_android/CHANGELOG.md:5]: Calls

(Please see the style guide linked from the PR checklist.)

File: packages/video_player/video_player_android/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ## NEXT
```
Comment (stuartmorgan) [inline packages/video_player/video_player_android/CHANGELOG.md:1]: Please see the docs linked from the checklist on how to update a changelog containing NEXT.

File: packages/video_player/video_player_android/CHANGELOG.md (hunk 1 above)
Comment (0xZOne) [reply packages/video_player/video_player_android/CHANGELOG.md]: done.

File: packages/video_player/video_player_android/CHANGELOG.md (hunk 2 above)
Comment (0xZOne) [reply packages/video_player/video_player_android/CHANGELOG.md]: done.

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Review by camsim99 — APPROVED** [review]
LGTM

**Pull request merged**